lparchive2epub: init at 1.2.3

peeley opened the pull request:
This PR adds a package for [lparchive2epub](https://github.com/Arwalk/lparchive2epub), a tool for downloading and converting [LP Archive](https://lparchive.org/) threads to EPUB files. 

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

peeley committed:
lparchive2epub: init at 1.2.4

FliegendeWurst commented on pkgs/by-name/lp/lparchive2epub/package.nix:
```diff
@@ -0,0 +1,35 @@
+    aiohttp
+  ];
+
+  meta = with lib; {
+    description = "Transform any LP from lparchive into an epub document.";
+    homepage = "https://github.com/Arwalk/lparchive2epub";
+    license = with licenses; [ mit ];
+    maintainers = with maintainers; [ nsnelson ];
```
```suggestion
  meta = {
    description = "Transform any LP from lparchive into an epub document";
    homepage = "https://github.com/Arwalk/lparchive2epub";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ nsnelson ];
```

FliegendeWurst commented on pkgs/by-name/lp/lparchive2epub/package.nix:
```diff
@@ -0,0 +1,35 @@
+  python3,
+}:
+
+python3.pkgs.buildPythonApplication rec {
```
I think python3Packages is preferred over python3.pkgs

peeley commented:
Apologies for the long turnaround!

Fixed the formatting of the `meta` field, and changed `python3.pkgs` to `python3Packages`. I also added `pytestCheckHook`, as tests were added/fixed upstream.
